### PR TITLE
Feature: new monitor type: MicroProfile Health Check

### LINKED ISSUE #2501: new monitor type: MicroProfile Health Check
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Monitor

### 🔖 Feature description

Adding "MicroProfile Health Check" as a new "specific monitor type".
The HTTP Request responds with a JSON, representing an application server with deployed applications. The response can be evaluated to get a detailed Overview of possible problems and their root cause.

### ✔️ Solution

I have implemented the feature and will provide a pull request shortly for further inspection.

### ❓ Alternatives

I have considered using the "HTTP keyword" monitor but in case of a "DOWN" not enough information can be extracted so extra steps are necessary to obtain the real cause. I want to shortcut these extra steps to improve response time.

### 📝 Additional Context

from [Eclipse MicroProfile Health](https://github.com/eclipse/microprofile-health/):
### Motivation
Health checks are used to probe the state of a computing node from another machine (i.e. kubernetes service controller) with the primary target being cloud infrastructure environments where automated processes maintain the state of computing nodes.

In this scenario, health checks are used to determine if a computing node needs to be discarded (terminated, shutdown) and eventually replaced by another (healthy) instance.

It’s not intended (although could be used) as a monitoring solution for human operators.

[Eclipse MicroProfile Health - Protocol and Wireformat](https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/protocol-wireformat.asciidoc#protocol-and-wireformat)

### PR DESCRIPTION
**Pull request by skaempfe**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes #2501 - new monitor type: MicroProfile Health Check

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by Computroniks — APPROVED** [review]
Looks good. Very nicely commented and easy to understand :+1:

**Comment by louislam** [comment]
MicroProfile is new to me, I would like to know are there any MicroProfile docker image that could be used to test this pr?

**Comment by skaempfe** [comment]
Hey @louislam , not sure how such a test setup should look like, have no experience in that topic.
What I can tell you: in the UP situation the target system will respond with an HTTP 200 and wil provide a JSON like this one (I added also the headers):
`*   Trying 127.0.0.1:9990...
* Connected to 127.0.0.1 (127.0.0.1) port 9990 (#0)
> GET /health HTTP/1.1
> Host: 127.0.0.1:9990
> User-Agent: curl/7.86.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Connection: keep-alive
< Content-Type: application/json
< Content-Length: 313
< Date: Thu, 23 Feb 2023 21:50:55 GMT
<
* Connection #0 to host 127.0.0.1 left intact
{"status":"UP","checks":[{"name":"server-state","status":"UP","data":{"value":"running"}},{"name":"empty-readiness-checks","status":"UP"},{"name":"empty-liveness-checks","status":"UP"},{"name":"deployments-status","status":"UP"},{"name":"empty-startup-checks","status":"UP"},{"name":"boot-errors","status":"UP"}]}`

In case of a Problem the remote system will respond with a HTTP status Code 503 and with a JSON like this one, detailing which component(s) caused the problem(s):
`*   Trying 127.0.0.1:9990...
* Connected to 127.0.0.1 (127.0.0.1) port 9990 (#0)
> GET /health HTTP/1.1
> Host: 127.0.0.1:9990
> User-Agent: curl/7.86.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 503 Service Unavailable
< Connection: keep-alive
< Content-Type: application/json
< Content-Length: 397
< Date: Thu, 23 Feb 2023 21:56:39 GMT
<
* Connection #0 to host 127.0.0.1 left intact
{"status":"DOWN","checks":[{"name":"server-state","status":"UP","data":{"value":"running"}},{"name":"deployments-status","status":"UP","data":{"department-service.war":"OK"}},{"name":"boot-errors","status":"UP"},{"name":"started-deployment.department-service.war","status":"UP"},{"name":"pl.piomin.services.department.health.ReadinessEndpoint","status":"DOWN"},{"name":"Server up","status":"UP"}]}`

maybe for the test case it might be sufficient to provide these two JSON together with their corresponding HTTP Status Code and the Content-Type header.

File: server/model/monitor.js [hunk 1]
```diff
@@ -910,6 +912,150 @@ class Monitor extends BeanModel {
...
+     * @param {Bean<heartbeat>} bean Heartbeat bean containing the current status
+     * @param {AxiosResponse<any>} res HTTP response containing the health check result
+     */
+    async analyzeHealthcheckResponse(bean, res) {
+        log.debug("monitor", "analyzeHealthcheckResponse");
+
+        const contentType = res.headers["content-type"];
+        if (! (contentType.startsWith("application/") && contentType.includes("json"))) {
+            bean.msg = `PENDING: unknown response content type: ${contentType}`;
+            bean.status = PENDING;
```
Comment (louislam) [inline server/model/monitor.js:926]: PENDING is used for `Retries` only, it should change to `DOWN`

**Comment by louislam** [comment]
I meant I want to do a test manually, but I don't know how to get it working. 

What I had tried to do:
1. Download a demo project from: https://start.microprofile.io/index.xhtml
2. Run `./gradlew libertyDev`
image
3. I placed the endpoint url into Uptime Kuma, but it didn't work.
image

**Pull request closed without merging**